CMakeLists.txt

```diff
@@ -1,39 +1,76 @@
 cmake_minimum_required(VERSION 2.8)
 
-set(PROJ_CJSON cJSON)
+set(CJSON_VERSION_MAJOR 1)
+set(CJSON_VERSION_MINOR 0)
+set(CJSON_VERSION_PATCH 0)
+set(CJSON_VERSION_SO 1)
+set(CJSON_UTILS_VERSION_SO 1)
+set(CJSON_VERSION "${CJSON_VERSION_MAJOR}.${CJSON_VERSION_MINOR}.${CJSON_VERSION_PATCH}")
+
+add_compile_options(-ansi -pedantic -Wall -Wextra -Werror -Wstrict-prototypes -Wwrite-strings)
+
+#variables for pkg-config
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(libdir "${CMAKE_INSTALL_PREFIX}/lib")
+set(version ${CJSON_VERSION})
 
+option(BUILD_SHARED_LIBS "Build shared libraries" ON)
+
+#cJSON
+set(PROJ_CJSON cJSON)
+set(CJSON_LIB cjson)
 project(${PROJ_CJSON} C)
 
 file(GLOB HEADERS cJSON.h)
 set(SOURCES cJSON.c)
 
-add_library(${PROJ_CJSON} ${HEADERS} ${SOURCES})
+add_library(${CJSON_LIB} ${HEADERS} ${SOURCES})
 if (NOT WIN32)
-	target_link_libraries(${PROJ_CJSON} m)
+    target_link_libraries(${CJSON_LIB} m)
 endif()
 
-set(PROJ_CJSON_UTILS cJSON_utils)
+configure_file("${cJSON_SOURCE_DIR}/libcjson.pc.in"
+    "${cJSON_BINARY_DIR}/libcjson.pc" @ONLY)
 
+install(TARGETS ${CJSON_LIB} DESTINATION lib${LIB_SUFFIX})
+install(FILES cJSON.h DESTINATION include/cjson)
+install (FILES ${cJSON_BINARY_DIR}/libcjson.pc DESTINATION lib/pkgconfig)
+
+set_target_properties(${CJSON_LIB}
+    PROPERTIES
+        SOVERSION ${CJSON_VERSION_SO}
+        VERSION ${CJSON_VERSION})
+
+#cJSON_Utils
+set(PROJ_CJSON_UTILS cJSON_utils)
+set(CJSON_UTILS_LIB cjson_utils)
 project(${PROJ_CJSON_UTILS} C)
 
 file(GLOB HEADERS_UTILS cJSON_Utils.h)
 set(SOURCES_UTILS cJSON_Utils.c)
 
-add_library(${PROJ_CJSON_UTILS} ${HEADERS_UTILS} ${SOURCES_UTILS})
-target_link_libraries(${PROJ_CJSON_UTILS} ${PROJ_CJSON})
+add_library(${CJSON_UTILS_LIB} ${HEADERS_UTILS} ${SOURCES_UTILS})
+target_link_libraries(${CJSON_UTILS_LIB} ${CJSON_LIB})
+
+configure_file("${cJSON_utils_SOURCE_DIR}/libcjson_utils.pc.in"
```

Comment:
This is a bit confusing to me (just like two projcet() statements in CMakeLists.txt) as I never have seen a package with two pkg-config files. What is the reason for this? Is it common?

To me it means that cJSON is actually two packages. If that holds why not split it into two repositories as well?

I am not 100% sure why fellow developers chose to use cJSON, but I assume that it was because it was lightweight and small (at that time there no utils stuff).